HW02LAZY/LAZY/Lazy.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Runtime.InteropServices;
+
+namespace MyLazy;
+
+public interface ILazy<T> { T? Get(); }
+
+/// <summary>
+/// Thread non-safe realisation of built-in Lazy class
+/// </summary>
+/// <typeparam name="T">Type of supplier function returning value</typeparam>
+public class LazySingleThread<T> : ILazy<T>
+{
+    private bool IsCalculated = false;
+    private T? Result;
+    private Func<T> Supplier;
+
+    /// <summary>
+    /// Creates object of this class
+    /// </summary>
+    /// <param name="supplier">Function, providing calculation</param>
+    public LazySingleThread(Func<T> supplier)
+    {
+        Supplier = supplier;
+    }
+
+    /// <summary>
+    /// Calls for lazy initialised object. First time runs calculation
+    /// via supplier function. Following times returns precalculated value
+    /// </summary>
+    /// <returns>Result of running supplier function</returns>
+    public T? Get()
+    {
+        if (!IsCalculated)
+            Result = Supplier();
```

Comment:
рекомендуется писать с фигурными скобками:
```suggestion
        if (!IsCalculated) {
            Result = Supplier();
        }
```